### PR TITLE
MRG: Minor fixes for 3.7

### PR DESCRIPTION
**Pull request by larsoner**
Should get tests passing on 3.7, and enable one of our Travis runs to use 3.7.

### CODE REVIEW

**Comment by larsoner** [comment]
The segmentation faults I fought through here *might* relate to #5806.

But in any case, ready for review/merge if the CIs come back happy finally.

**Comment by agramfort** [comment]
thx @larsoner

File: conftest.py [hunk 1]
```diff
@@ -4,9 +4,19 @@
...
 import pytest
+import warnings
+# For some unknown reason, on Travis-xenial there are segfaults caused on
+# the line pytest -> pdb.Pdb.__init__ -> "import readline". Forcing an
+# import here seems to prevent them (!?). This suggests a potential problem
+# with some other library stepping on memory where it shouldn't. It only
+# seems to happen on the Linux runs that install Mayavi. Anectodally,
+# @larsoner has had problems a couple of years ago where a mayavi import
+# seemed to corrupt SciPy linalg function results (!), likely due to the
+# associated VTK import, so this could be another manifestation of that.
```
Comment (agramfort) [inline conftest.py]: do you think what you report with segfault can be related to #5806 ?

cc @yarikoptic @tillea

Comment (yarikoptic) [reply conftest.py]: nothing could be said certain whenever segfault is in question and no traceback info ;)  so it might or might not be

**Comment by agramfort** [comment]
let's say it's just a hunch as we have no other painful compiled code than
vtk...

Unfortunately I cannot replicated easily :(

File: conftest.py (hunk 1 above)
Comment (larsoner) [reply conftest.py]: It's possible, I'm going to look into it

**Pull request merged**